backend/runner/runner.go

```diff
@@ -298,3 +312,53 @@ func (s *Service) registrationLoop(ctx context.Context, send func(request *ftlv1
 	}
 	return nil
 }
+
+func (s *Service) streamLogsLoop(ctx context.Context, send func(request *ftlv1.StreamDeploymentLogsRequest) error) error {
+	delay := time.Millisecond * 500
+
+	deployment, ok := s.deployment.Load().Get()
+	if !ok {
+		// No deployment, nothing to do.
+		select {
+		case <-ctx.Done():
+			err := context.Cause(ctx)
+			return errors.WithStack(err)
+		case <-time.After(delay):
+			return nil
+		}
+	}
```

Comment:
My goal here is to try to get the deployment if there is one. If not then we just delay and check again after the delay. Not sure if there's a better pattern for this or not.

Reply:
I'd probably just load it in the `case entry` below, and drop the log entry if there's no deployment available.

Reply:
I would load it there, but since the log queue is on the `deployment` I need to load it first to get the `entry`. I could move the queue to the `Service` instead which would let me get access to it regardless of whether there is a deployment or not. It sort of feels like it's nice that this queue gets cleaned up when there's no deployment, but either way should work. I'll still with what I have now and we can update later if needed.

Reply:
Ah yeah, makes sense.